File: frontend/netlify/functions/upload-resume.js
exports.handler = async function(event, context) {
  // Handle CORS preflight
  if (event.httpMethod === 'OPTIONS') {
    return {
      statusCode: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Headers': 'Content-Type',
        'Access-Control-Allow-Methods': 'POST, OPTIONS'
      },
      body: ''
    };
  }

  // Only allow POST requests
  if (event.httpMethod !== 'POST') {
    return { statusCode: 405, body: 'Method Not Allowed' };
  }

  try {
    // For Netlify functions, file upload is complex
    // We'll return a simple response asking user to paste text directly
    return {
      statusCode: 200,
      headers: {
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Headers': 'Content-Type',
        'Access-Control-Allow-Methods': 'POST, OPTIONS'
      },
      body: JSON.stringify({
        text: "Please paste your resume text directly in the resume text area instead of uploading a file. File upload is not supported in this deployment.",
        message: "File upload not available - please paste resume text directly"
      })
    };

  } catch (error) {
    console.error('Error in upload-resume function:', error);
    return {
      statusCode: 500,
      body: JSON.stringify({ 
        detail: 'Internal server error',
        message: error.message 
      })
    };
  }
};

File: netlify/functions/upload-resume.py
import json
import base64
from app.services.file_parser import parse_resume

def handler(event, context):
    if event['httpMethod'] != 'POST':
        return {
            'statusCode': 405,
            'body': json.dumps({'error': 'Method not allowed'})
        }

    try:
        # Assuming the file is sent as base64 in body
        body = json.loads(event['body'])
        file_data = base64.b64decode(body['file'])
        filename = body['filename']
        text = parse_resume(file_data, filename)
        if not text.strip():
            return {
                'statusCode': 400,
                'body': json.dumps({'error': 'Could not extract text from the file.'})
            }
        return {
            'statusCode': 200,
            'headers': {'Content-Type': 'application/json'},
            'body': json.dumps({'text': text})
        }
    except ValueError as e:
        return {
            'statusCode': 400,
            'body': json.dumps({'error': str(e)})
        }
    except Exception as e:
        return {
            'statusCode': 500,
            'body': json.dumps({'error': f'Failed to parse file: {str(e)}'})
        }